cmd/test-bot.rb

```diff
@@ -71,6 +71,8 @@ def test_bot_args
              description: "Only run the formulae detection steps."
       switch "--only-formulae-dependents",
              description: "Only run the formulae dependents steps."
+      switch "--only-formulae-fetch",
```

Comment:
To be consistent with other options/behaviour: this needs to be run unconditionally unless this flag is passed. I think it would be good to have this split out into a separate job for general PRs, too (but not blocking)? More, smaller jobs is better, I think.

We could even perhaps get some decent optimisation by fetching the sources only on Linux, publishing as artifacts and using that to fan out the downloads onto all the other platforms?

Reply:
For now: might also want to clarify in the docs that this is an optional, post-upload fetch.

Reply:
Added a clarification!